File: server/src/services/projects-service/internal/types/tasks-types.go
package types

type OneProjectTasksResponse struct {
	ID          string `json:"id" db:"id"`
	Title       string `json:"title" db:"title"`
	Description string `json:"description" db:"description"`
	Status      string `json:"status" db:"status"`
	ProjectID   string `json:"project_id" db:"project_id"`
}

File: frontend/src/types/tasks-types.ts
export type Task = {
    id: string;
    title: string;
    description?: string;
    start_date: Date;
    end_date?: Date;
    status:
        | "not_started"
        | "in_progress"
        | "completed"
        | "on_hold"
        | "cancelled"
        | "blocked"
        | "review"
        | "testing"
        | "deployed"
        | "archived";
    user_id: string;
    project_id: string;
    created_at: Date;
    updated_at: Date;
};
